### PR TITLE
ICMSLST-1451 Test coverage for data_migration app

### PR DESCRIPTION
**Pull request by mikeylockwood**
```
Name                                                                  Stmts   Miss Branch BrPart  Cover   Missing
-----------------------------------------------------------------------------------------------------------------
data_migration/management/commands/export_from_v1.py                     72     10     32      1    80%   100, 114-143
data_migration/management/commands/import_v1_data.py                     46      0     18      0   100%
data_migration/management/commands/utils/db.py                           19      0      0      0   100%
data_migration/management/commands/utils/format.py                       15      0     10      0   100%
data_migration/models/base.py                                            50      1     20      0    99%   86
data_migration/models/import_application/import_application.py           97      8     18      0    90%   119-125, 129, 133
data_migration/models/import_application/import_application_type.py      60      0      8      0   100%
data_migration/models/import_application/quota.py                        63     14     10      0    81%   48-51, 74-75, 79-82, 94-97
data_migration/models/reference/reference.py                             62      2     20      0    98%   72, 90
data_migration/models/user/user.py                                       50      0      6      0   100%
-----------------------------------------------------------------------------------------------------------------
TOTAL                                                                   534     35    142      1    93%
                                                534     42    142      2    91%

```

### CODE REVIEW

**Comment by MattHolmes123** [comment]
Are the lines not covered not worth testing / hard to test?

File: data_migration/tests/test_export.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+
+@override_settings(ALLOW_DATA_MIGRATION=True)
+@override_settings(APP_ENV="production")
+@override_settings(ICMS_PROD_USER="")
+@override_settings(ICMS_PROD_PASSWORD="1234")
+@pytest.mark.django_db
+@mock.patch.object(cx_Oracle, "connect")
+def test_create_user_no_username(mock_connect):
+    with pytest.raises(CommandError) as exc_info:
```
Comment (MattHolmes123) [inline data_migration/tests/test_export.py:36]: You can do the match in the raises if you want.
e.g. 
```
with pytest.raises(CommandError match="No user details found for this environment")
```

File: data_migration/tests/test_import.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+def test_data_import_not_enabled_non_prod():
+    with pytest.raises(CommandError) as exc_info:
+        call_command("import_v1_data")
+    assert exc_info.match("Data migration has not been enabled for this environment")
+
+
+@override_settings(ALLOW_DATA_MIGRATION=True)
+@override_settings(APP_ENV="production")
+@pytest.mark.django_db
+@mock.patch.dict(DATA_TYPE_SOURCE_TARGET, {"reference": [(C, Country), (CG, CountryGroup)]})
```
Comment (MattHolmes123) [inline data_migration/tests/test_import.py:34]: This is cool, didn't know you could patch variables (#noob)

Comment (mikeylockwood) [reply data_migration/tests/test_import.py:34]: Me neither, made it much easier when I found out how

**Comment by mikeylockwood** [comment]
> Are the lines not covered not worth testing / hard to test?

Basically yeah. I would have to add a lot of test data to fully cover the models and all it really would be testing is the methods that return static lists.

For the export_v1_data, a lot of test data would have to be added again for the import applications and the majority of it has already been tested in the utils files.

**Review by MattHolmes123 — APPROVED** [review]

**Pull request merged**